### PR TITLE
chore: add ability to strip prerelease information from the version

### PR DESCRIPTION
**Pull request by garethjevans**
Add a flag to strip the prerelease information from the determined version before bumping. e.g.

Before:

`1.0.0-something` -> `1.0.0`

After (with the flag set)

`1.0.0-something` -> `1.0.1`

This behaviour will make prerelease act more like build metadata with the advantage that build metadata cannot be added to a docker image version, but prerelease can.

### CODE REVIEW

**Comment by vbehar** [comment]
nice, thanks for the PR.
because this only applies to the `semantic` strategy, what do you think about passing this flag along with the strategy directly? such as:

```
jx-release-version -next-version=semantic:strip-prerelease
```

(or `semantic:without-prerelease`)

similar to what we're already doing with other strategies such as

```
jx-release-version -previous-version=from-file:pom.xml
```

to avoid having too many root flags which only applies to a specific use-case

**Comment by garethjevans** [comment]
How would multiple values work in this case? e.g.

`-next-version=from-file:pom.xml,strip-prerelease` - would that be valid?

**Comment by vbehar** [comment]
well, for this use-case, if you read the next version from a file and want to strip the prerelease, you can do it with the output format, something like `-next-version=from-file:pom.xml -output-format={{.Major}}.{{.Minor}}.{{.Patch}}`

but yes we can support multiple values in the strategy args here https://github.com/jenkins-x-plugins/jx-release-version/blob/master/main.go#L125-L134 or pass the whole arg to the strategy and let it re-parse it as it wants.

so basically I think:
- if we only need this for the semantic strategy, maybe using a strategy arg is a good option
- if we want to be able to support it for any strategy, maybe we can introduce some kind of processor between previous-version and next-version?

**Review by vbehar — APPROVED** [review]

**Comment by garethjevans** [comment]
Ah cool, I've pushed a PR that should support this.  I realised my previous comment was wrong so I removed it (apologies - i didn't realise you were replying to it!)
